### ISSUE TITLE
Router Implementation in Synthetic Experiments

### ISSUE THREAD

**Issue by ZichenMiao** [uclaml/MoE#2]
Hi,

Thanks for this great work!!

I'm curious about one implementation in the synthetic experiments, where router probability is directly set to logits (i.e., the output of the convolutional layer), without passing the softmax function. This is different from the router definition in the paper and also other popular router implementations.

Btw, I have tried adding the softmax function back to the router and the results look fine.

(code snippet copied from MoE_Visualization.ipynb)
```
class Router(nn.Module):
    def __init__(self, input_dim, out_dim, patch_num, strategy='top1',noise=None):
        super(Router, self).__init__()
        self.conv1 = nn.Conv1d(1, out_dim, int(input_dim/patch_num), int(input_dim/patch_num),bias=False)
        self.out_dim = out_dim
        self.strategy = strategy
        if self.strategy == 'top1':
            if noise is None:
                self.noise = torch.rand(DATA_NUM,out_dim).cuda()
            else:
                self.noise = noise
        # zero initialization
        self.reset_parameters()
    
    def reset_parameters(self):
        self.conv1.weight = torch.nn.Parameter(self.conv1.weight * 0)
            
    def forward(self, x):      
        x = self.conv1(x)
        x = torch.sum(x,2)
        if self.strategy == 'top1':
            output = x + self.noise
        ## with softmax
        ## output = torch.softmax(output, dim=-1)
        return output
```
